### PR TITLE
Head request for file info

### PR DESCRIPTION
**Pull request by dlockhart**
@omsmith, @jstefaniuk-d2l: OKish? Going to write a bunch of tests around the `fileInfoProvider` (if Jest will let me) and add in some nicer error handling in the UI if the XHR fails.

### CODE REVIEW

File: src/fileInfoProvider.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		if (xhr.readyState !== 4) {
+			return;
+		}
+		var size = xhr.getResponseHeader('content-length');
+		if (size === null) {
+			return callback(new Error('Missing "content-length" header'));
+		}
+		var mimeType = xhr.getResponseHeader('content-type');
+		if (mimeType === null) {
+			return callback(new Error('Missing "content-type" header'));
```
Comment (jstefaniuk-d2l) [inline src/fileInfoProvider.js]: Will there ever be the case where we would want to show a file that had no mimeType? (or is that not possible?)

Comment (omsmith) [inline src/fileInfoProvider.js]: "unknown mime-type" tends to -> "application/octet-stream" ("these are bytes")

although i believe d2l does "d2l/unknowntype"

File: src/fileViewer.js [hunk 2]
```diff
@@ -1,33 +1,49 @@
...
+	componentDidMount: function() {
+		this.fetchFileInfo(this.props.src);
+	},
+	componentWillReceiveProps: function(nextProps) {
+		if (nextProps.src !== this.props.src) {
+			this.fetchFileInfo(nextProps.src);
+		}
+	},
+	fetchFileInfo: function(src) {
+		var me = this;
```
Comment (jstefaniuk-d2l) [inline src/fileViewer.js]: I'm just wondering what the point of this is?

Comment (omsmith) [inline src/fileViewer.js]: In the callback below (in the call to `fileInfoProvider`) we lose the calling context, so we assign it to a variable so we can reference it there (`me.setState`)

File: src/fileViewer.js [hunk 3]
```diff
@@ -1,33 +1,49 @@
...
+		this.setState({error: null, info: null});
+		fileInfoProvider(src, function(err, fileInfo) {
+			if (err) {
+				me.setState({error: err});
+				return;
+			}
+			me.setState({info: fileInfo});
```
Comment (omsmith) [inline src/fileViewer.js]: Should this call (and the one above) be checking for `isMounted` ?

File: src/fileViewer.js (hunk 2 above)
Comment (dlockhart) [inline src/fileViewer.js]: Yeah, when you're inside an anonymous function, `this` refers to the function, so we assign `this` to a variable in the outer scope. It's weird I know... JavaScript.

File: src/fileInfoProvider.js (hunk 1 above)
Comment (dlockhart) [inline src/fileInfoProvider.js]: Cool I'll switch this to fall back to that then instead of blowing up.

File: src/fileViewer.js (hunk 3 above)
Comment (dlockhart) [inline src/fileViewer.js]: So I can, although it's only called from `componentDidMount` (which happens after mounting) and `componentWillReceiveProps` which again can only happen after mounting.

Comment (omsmith) [inline src/fileViewer.js]: Right, but if it's async it could have unmounted in the meantime

Comment (omsmith) [inline src/fileViewer.js]: That's the wisdom this page goes for, anyway, https://facebook.github.io/react/tips/initial-ajax.html

Comment (dlockhart) [inline src/fileViewer.js]: Ah gotcha, after the asynch returns. Will fix, thx.

**Comment by omsmith** [comment]
Looks good

**Pull request merged**